server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -167,19 +167,21 @@ class InteractiveSession(
       _state match {
         case SessionState.Idle() =>
           _state = SessionState.Busy()
-
-          Http(svc.DELETE OK as.String).either() match {
-            case (Right(_) | Left(_: ConnectException)) =>
-              // Make sure to eat any connection errors because the repl shut down before it sent
-              // out an OK.
+          Future {
+            try {
+              new AsyncHttpClient().prepareDelete(url.get.toString).execute()
               synchronized {
                 _state = SessionState.Dead()
               }
-
               Future.successful(())
-
-            case Left(t: Throwable) =>
-              Future.failed(t)
+            } catch {
+              case e: ConnectException =>
+                synchronized {
```

Comment:
It shouldn't be needed to synchronize here; the synchronization in the rest of this class is kinda spotty already, it would be better to fix all of it in a separate change.